pulser-simulation/pulser_simulation/qutip_config.py

```diff
@@ -0,0 +1,127 @@
+
+import numpy as np
+
+from pulser.backend.config import EmulationConfig
+from pulser_simulation.qutip_state import QutipState
+
+
+class QutipConfig(EmulationConfig[QutipState]):
+    """The configuration of a QutipBackend.
+
+    Args:
+        observables: A sequence of observables to compute at specific
+            evaluation times. The observables without specified evaluation
+            times will use this configuration's 'default_evaluation_times'.
+        sampling_rate: The fraction of samples to extract from the pulse
+            sequence for emulation.
+        default_evaluation_times: The default times at which observables
+            are computed. Can be a sequence of relative times between 0
+            (the start of the sequence) and 1 (the end of the sequence).
+            Can also be specified as "Full", in which case every step in the
+            emulation will also be an evaluation time.
+        initial_state: The initial state from which emulation starts. If
+            specified, the state type needs to be compatible with the emulator
+            backend. If left undefined, defaults to starting with all qudits
+            in the ground state.
+        with_modulation: Whether to emulate the sequence with the programmed
+            input or the expected output.
+        prefer_device_noise_model: If the sequence's device has a default noise
+            model, this option signals the backend to prefer it over the noise
+            model given with this configuration.
+        noise_model: An optional noise model to emulate the sequence with.
+            Ignored if the sequence's device has default noise model and
+            `prefer_device_noise_model=True`.
+
+    See Also:
+        EmulationConfig: The base configuration class for an EmulatorBackend.
+    """
+
+    sampling_rate: float
+
+    def __init__(
+        self,
+        *,
+        sampling_rate: float = 1.0,
+        default_evaluation_times: Sequence[float] | Literal["Full"] = "Full",
+        **backend_options: Any,
+    ):
+        """Initializes a QutipConfig."""
+        if backend_options.setdefault("interaction_matrix"):
+            raise NotImplementedError(
+                "'QutipBackendV2' does not handle custom interaction matrices."
+            )
+        if not (0 < sampling_rate <= 1.0):
+            raise ValueError(
+                f"The sampling rate (`sampling_rate` = {sampling_rate}) must"
+                " be greater than 0 and less than or equal to 1."
+            )
+        initial_state = backend_options.setdefault("initial_state")
+        if initial_state and not isinstance(initial_state, QutipState):
+            raise TypeError(
+                "If provided, `initial_state` must be an instance of "
+                f"`QutipState`, not {type(initial_state)}."
+            )
+        super().__init__(
+            sampling_rate=sampling_rate,
+            default_evaluation_times=default_evaluation_times,
```

Comment:
I'm a bit confused about what's happening with `default_evaluation_times`. It exists in `EmulationConfig`, but there it's `np.ndarray` rather than `Sequence[float]`, so that could be the reason to redefine it here, but then I would expect some logic to cast it to `ndarray` before feeding it to `super`